### PR TITLE
Added SQL statement for outbox cleanup in scaled environments

### PR DESCRIPTION
**Pull request by poornimanayar**

### CODE REVIEW

File: persistence/sql/outbox.md [hunk 1]
```diff
@@ -27,4 +27,9 @@ In scaled-out environments, endpoint instances compete to execute outbox cleanup
 
 - Run the cleanup on only a single instance.
 - Increase the cleanup interval so that, on average, one endpoint instance cleans up as often as a single instance would normally. The cleanup timer isn't strict and over time these will drift and will cause less overlap. For example, for 10 endpoint instances, let cleanup run every 10 minutes instead of every minute.
-- Disable cleanup on all instances and have cleanup run as a scheduled job in the database.
+Disable cleanup on all instances and have cleanup run as a scheduled job in the database. The following SQL statement can be used for this purpose. It is important to ensure that the value of `@DispatchedBefore` is the same as the value of the deduplication window.
+  ```sql
+  delete top (@BatchSize) from {tableName} with (rowlock) 
+  where (Dispatched = 1 and DispatchedAt < @DispatchedBefore) 
+  or (Dispatched = 0 and Operations = '[]' and DispatchedAt IS NULL);
```
Comment (andreasohlund) [inline persistence/sql/outbox.md:34]: This part is the workaround for the bug, not sure if we can mention this here unless we check that the index use is performant enough for all users?

Also this page is generic for all supported dialects and eg. Postgres use a different query 

https://github.com/Particular/NServiceBus.Persistence.Sql/blob/master/src/SqlPersistence/Outbox/SqlDialect_PostgreSql.cs#L76

In short; If we go down this path I think we need to have dedicated sections for each supported dialect

As for the workaround: It's already documented here https://github.com/Particular/NServiceBus.TransactionalSession/issues/376

**Comment by poornimanayar** [comment]
@andreasohlund @lailabougria ready to be reviewed.

File: persistence/sql/outbox.md [hunk 2]
```diff
@@ -27,4 +27,8 @@ In scaled-out environments, endpoint instances compete to execute outbox cleanup
 
 - Run the cleanup on only a single instance.
 - Increase the cleanup interval so that, on average, one endpoint instance cleans up as often as a single instance would normally. The cleanup timer isn't strict and over time these will drift and will cause less overlap. For example, for 10 endpoint instances, let cleanup run every 10 minutes instead of every minute.
-- Disable cleanup on all instances and have cleanup run as a scheduled job in the database.
+- Disable cleanup on all instances and have cleanup run as a scheduled job in the database. See the dialect-specific scripts for
+  - [SQL Server](https://docs.particular.net/persistence/sql/sqlserver-scripts#run-time-outbox)
+  - [MySQL](https://docs.particular.net/persistence/sql/mysql-scripts#run-time-outbox)
+  - [Oracle](https://docs.particular.net/persistence/sql/oracle-scripts#run-time-outbox)
+  - [PostgreSQL](https://docs.particular.net/persistence/sql/postgresql-scripts#run-time-outbox)
```
Comment (andreasohlund) [inline persistence/sql/outbox.md:34]:
```suggestion
  - [SQL Server](/persistence/sql/sqlserver-scripts.md#run-time-outbox)
  - [MySQL](/persistence/sql/mysql-scripts.md#run-time-outbox)
  - [Oracle](/persistence/sql/oracle-scripts.md#run-time-outbox)
  - [PostgreSQL](/persistence/sql/postgresql-scripts.md#run-time-outbox)
```

**Review by andreasohlund — APPROVED** [review]

File: persistence/sql/outbox.md [hunk 3]
```diff
@@ -27,4 +27,8 @@ In scaled-out environments, endpoint instances compete to execute outbox cleanup
 
 - Run the cleanup on only a single instance.
 - Increase the cleanup interval so that, on average, one endpoint instance cleans up as often as a single instance would normally. The cleanup timer isn't strict and over time these will drift and will cause less overlap. For example, for 10 endpoint instances, let cleanup run every 10 minutes instead of every minute.
-- Disable cleanup on all instances and have cleanup run as a scheduled job in the database.
+- Disable cleanup on all instances and have cleanup run as a scheduled job in the database. See the dialect-specific scripts for
```
Comment (lailabougria) [inline persistence/sql/outbox.md:30]:
```suggestion
- Disable cleanup on all instances and run cleanup as a scheduled job in the database. See the dialect-specific scripts for more details:
```

**Review by lailabougria — APPROVED** [review]

**Pull request merged**